x-pack/plugins/alerting/server/application/rule/methods/resolve/resolve_rule.ts

```diff
@@ -7,20 +7,20 @@
 
 import { AlertConsumers } from '@kbn/rule-data-utils';
 
-import { RawRule, RuleTypeParams, ResolvedSanitizedRule } from '../../types';
-import { ReadOperations, AlertingAuthorizationEntity } from '../../authorization';
-import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
-import { getAlertFromRaw } from '../lib/get_alert_from_raw';
-import { RulesClientContext } from '../types';
-import { formatLegacyActions } from '../lib';
+import { ruleAuditEvent, RuleAuditAction } from '../../../../rules_client/common/audit_events';
+import { RawRule, RuleTypeParams, ResolvedSanitizedRule } from '../../../../types';
+import { ReadOperations, AlertingAuthorizationEntity } from '../../../../authorization';
+import { getAlertFromRaw } from '../../../../rules_client/lib/get_alert_from_raw';
```

Comment:
In general, we want to move away from the `RawRule` types and use the new `Rule/RuleDomain/RuleAttribute` types in the rules client methods. `RuleDomain` is analogous to the old `Rule` type, it has all of the rule properties, whereas `Rule` is analogous to the `SanitizedRule` type as we strip out certain attributes as this type is shared between solutions.

The reasoning is these new types are now explicitly defined in that they are not derived from other types (something we did quite a bit with the existing `sanitizedRule` for example). In fact, they are derived from schemas.

We also want to have explicit transformations between the `application` and the `data` layer (`RuleDomain` -> `RuleAttributes` types using the new transform functions).

So the workflow would look like this:

1. Call resolveRule
2. Validate input params using schemas
3. Call resolve -> returns the saved object data object (`RuleAttributes`)
4. Transform the `RuleAttributes` data layer type to `RuleDomain` application layer type using `transformRuleAttributesToRuleDomain` (analogous to `getAlertFromRaw` but uses the new types). 
5. Validate the output rule using schemas
6. Transform the `RuleDomain` to `Rule` since the original function returns a `SanitizedRule`, done using `transformRuleDomainToRule`

So the only case where we would still use `SanitizedRule` is to maintain compatibility with parts of the code base that have not been converted yet. Feel free to take a look at `create_rule.ts` for an example.